qiskit/compiler/transpiler.py

```diff
@@ -650,16 +711,12 @@ def _parse_basis_gates(basis_gates, backend, circuits):
                 basis_gates = getattr(backend.configuration(), "basis_gates", None)
         else:
             basis_gates = backend.operation_names
-    # basis_gates could be None, or a list of basis, e.g. ['u3', 'cx']
-    if basis_gates is None or (
-        isinstance(basis_gates, list) and all(isinstance(i, str) for i in basis_gates)
-    ):
-        basis_gates = [basis_gates] * len(circuits)
-
+    if basis_gates is not None:
+        basis_gates = list(basis_gates)
```

Comment:
What is this `list` call for?  Is it normalisation of an iterable? I originally thought it was about not passing a mutable object through to the passes, but the previous version of this code did that if the `transpile` ran in serial, seemingly without problems.

Reply:
I don't really remember tbh, I do think I had to add this because there was something that failed without it. But this also could be leftover from me debugging the case where prior to this PR a user passed `basis_gates=['u', 'cx']` over two circuits each circuit got a single basis gate. I'll try to remove it locally and see what happens.